[C++][Restbed-server] Compilation error from missing function call parameter

<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description
The following compilation error occurs when I'm trying to compile a Restbed server from the given specification:
```
/api/FrameApi.cpp:70:91: error: expected primary-expression before ‘)’ token
    const bool includePictureBase64 = request->get_query_parameter("includePictureBase64", );
```
<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

##### openapi-generator version
3.2.1(latest master branch with PR #752)
<!-- which version of openapi-generator are you using, is it a regression? -->

##### OpenAPI declaration file content or url
https://raw.githubusercontent.com/deepviss/deepviss-server-example/master/src/main/resources/deep-viss.json
<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

##### Command line used for generation
Inside cloned repository, after mvn clean package command:
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate/
        -i deep-viss.json /
        -g cpp-restbed-server /
        -o ../openapi-restbed-server
```
<!-- including the language, libraries and various options -->

##### Steps to reproduce
Assuming Restbed is installed on local machine https://github.com/Corvusoft/restbed
I used the following cmake file:
```
cmake_minimum_required(VERSION 3.10)
project(restbed_ex)

set(CMAKE_CXX_STANDARD 11)

file(GLOB_RECURSE src_files mode/*.cpp api/*.cpp)

add_executable(restbed_ex main.cpp ${src_files})
target_include_directories(restbed_ex PRIVATE /restbed/include/)
target_include_directories(restbed_ex PRIVATE model/)
target_include_directories(restbed_ex PRIVATE api/)
target_link_libraries(restbed_ex restbed)
```
1)Go to output folder
2)Make a dummy main.cpp (touch main.cpp)
3)mkdir build && cd build
4)cmake ..
5)make
<!-- unambiguous set of steps to reproduce the bug.-->

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
@@ -2742,9 +2742,8 @@ public class DefaultCodegen implements CodegenConfig {
             }
 
             // set default value
-            if (parameterSchema.getDefault() != null) {
-                codegenParameter.defaultValue = toDefaultValue(parameterSchema);
-            }
+            codegenParameter.defaultValue = toDefaultValue(parameterSchema);
+
             // TDOO revise collectionFormat
             String collectionFormat = null;
             if (ModelUtils.isArraySchema(parameterSchema)) { // for array parameter

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestbedServerCodegen.java
@@ -25,14 +25,18 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.logging.Logger;
 
 import org.apache.commons.lang3.StringUtils;
 import org.openapitools.codegen.*;
 import org.openapitools.codegen.utils.ModelUtils;
 
 import io.swagger.v3.oas.models.media.*;
+import org.slf4j.LoggerFactory;
 
 public class CppRestbedServerCodegen extends AbstractCppCodegen {
+
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(CppRestbedServerCodegen.class);
 
     public static final String DECLSPEC = "declspec";
     public static final String DEFAULT_INCLUDE = "defaultInclude";
@@ -287,25 +291,63 @@ public class CppRestbedServerCodegen extends AbstractCppCodegen {
     @Override
     public String toDefaultValue(Schema p) {
         if (ModelUtils.isStringSchema(p)) {
-            return "\"\"";
+            if (p.getDefault() != null) {
+                return "\"" + p.getDefault().toString() + "\"";
+            } else {
+                return "\"\"";
+            }
         } else if (ModelUtils.isBooleanSchema(p)) {
-            return "false";
+            if (p.getDefault() != null) {
+                return p.getDefault().toString();
+            } else {
+                return "false";
+            }
         } else if (ModelUtils.isDateSchema(p)) {
-            return "\"\"";
+            if (p.getDefault() != null) {
+                return "\"" + p.getDefault().toString() + "\"";
+            } else {
+                return "\"\"";
+            }
         } else if (ModelUtils.isDateTimeSchema(p)) {
-            return "\"\"";
+            if (p.getDefault() != null) {
+                return "\"" + p.getDefault().toString() + "\"";
+            } else {
+                return "\"\"";
+            }
         } else if (ModelUtils.isNumberSchema(p)) {
-            if (ModelUtils.isFloatSchema(p)) {
-                return "0.0f";
+            if (ModelUtils.isFloatSchema(p)) { // float
+                if (p.getDefault() != null) {
+                    return p.getDefault().toString() + "f";
+                } else {
+                    return "0.0f";
+                }
+            } else { // double
+                if (p.getDefault() != null) {
+                    return p.getDefault().toString();
+                } else {
+                    return "0.0";
+                }
             }
-            return "0.0";
         } else if (ModelUtils.isIntegerSchema(p)) {
-            if (ModelUtils.isLongSchema(p)) {
-                return "0L";
+            if (ModelUtils.isLongSchema(p)) { // long
+                if (p.getDefault() != null) {
+                    return p.getDefault().toString() + "L";
+                } else {
+                    return "0L";
+                }
+            } else { // integer
+                if (p.getDefault() != null) {
+                    return p.getDefault().toString();
+                } else {
+                    return "0";
+                }
             }
-            return "0";
         } else if (ModelUtils.isByteArraySchema(p)) {
-            return "\"\"";
+            if (p.getDefault() != null) {
+                return "\"" + p.getDefault().toString() + "\"";
+            } else {
+                return "\"\"";
+            }
         } else if (ModelUtils.isMapSchema(p)) {
             String inner = getSchemaType(ModelUtils.getAdditionalProperties(p));
             return "std::map<std::string, " + inner + ">()";
@@ -319,6 +361,7 @@ public class CppRestbedServerCodegen extends AbstractCppCodegen {
         } else if (!StringUtils.isEmpty(p.get$ref())) {
             return "new " + toModelName(ModelUtils.getSimpleRef(p.get$ref())) + "()";
         }
+
         return "nullptr";
     }
 

--- a/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache
@@ -75,7 +75,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMet
 			// Getting the path params
 			{{#pathParams}}
 			{{#isPrimitiveType}}
-			const {{{dataType}}} {{{paramName}}} = request->get_path_parameter("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_path_parameter("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/pathParams}}
 			{{/hasPathParams}}
@@ -84,7 +84,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMet
 			// Getting the query params
 			{{#queryParams}}
 			{{#isPrimitiveType}}
-			const {{{dataType}}} {{{paramName}}} = request->get_query_parameter("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_query_parameter("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/queryParams}}
 			{{/hasQueryParams}}
@@ -93,7 +93,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMet
 			// Getting the headers
 			{{#headerParams}}
 			{{#isPrimitiveType}}
-			const {{{dataType}}} {{{paramName}}} = request->get_header("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_header("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/headerParams}}
 			{{/hasHeaderParams}}
@@ -140,7 +140,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMet
 			// Getting the path params
 			{{#pathParams}}
 			{{#isPrimitiveType}}
-			const {{dataType}} {{paramName}} = request->get_path_parameter("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_path_parameter("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/pathParams}}
 			{{/hasPathParams}}
@@ -149,16 +149,16 @@ void {{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMet
 			// Getting the query params
 			{{#queryParams}}
 			{{#isPrimitiveType}}
-			const {{dataType}} {{paramName}} = request->get_query_parameter("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_query_parameter("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/queryParams}}
 			{{/hasQueryParams}}
-		
+
 			{{#hasHeaderParams}}
 			// Getting the headers
 			{{#headerParams}}
 			{{#isPrimitiveType}}
-			const {{dataType}} {{paramName}} = request->get_header("{{paramName}}", {{#isString}}""{{/isString}}{{#isInteger}}0{{/isInteger}}{{#isLong}}0L{{/isLong}}{{#isFloat}}0.0f{{/isFloat}}{{#isDouble}}0.0{{/isDouble}});
+			const {{{dataType}}} {{{paramName}}} = request->get_header("{{paramName}}", {{{defaultValue}}});
 			{{/isPrimitiveType}}
 			{{/headerParams}}
 			{{/hasHeaderParams}}

--- a/samples/server/petstore/cpp-restbed/api/PetApi.cpp
+++ b/samples/server/petstore/cpp-restbed/api/PetApi.cpp
@@ -118,7 +118,7 @@ void PetApiPetResource::PUT_method_handler(const std::shared_ptr<restbed::Sessio
 			std::string requestBody = restbed::String::format("%.*s\n", ( int ) body.size( ), body.data( ));
 
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;
@@ -194,7 +194,7 @@ void PetApiPetPetIdResource::GET_method_handler(const std::shared_ptr<restbed::S
 			// Getting the path params
 			const int64_t petId = request->get_path_parameter("petId", 0L);
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;
@@ -225,7 +225,7 @@ void PetApiPetPetIdResource::POST_method_handler(const std::shared_ptr<restbed::
 			// Getting the path params
 			const int64_t petId = request->get_path_parameter("petId", 0L);
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;

--- a/samples/server/petstore/cpp-restbed/api/StoreApi.cpp
+++ b/samples/server/petstore/cpp-restbed/api/StoreApi.cpp
@@ -100,7 +100,7 @@ void StoreApiStoreOrderOrderIdResource::GET_method_handler(const std::shared_ptr
 			// Getting the path params
 			const int64_t orderId = request->get_path_parameter("orderId", 0L);
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;

--- a/samples/server/petstore/cpp-restbed/api/UserApi.cpp
+++ b/samples/server/petstore/cpp-restbed/api/UserApi.cpp
@@ -253,7 +253,7 @@ void UserApiUserUsernameResource::GET_method_handler(const std::shared_ptr<restb
 			// Getting the path params
 			const std::string username = request->get_path_parameter("username", "");
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;
@@ -292,7 +292,7 @@ void UserApiUserUsernameResource::PUT_method_handler(const std::shared_ptr<restb
 			// Getting the path params
 			const std::string username = request->get_path_parameter("username", "");
 			
-		
+
 			
 			// Change the value of this variable to the appropriate response before sending the response
 			int status_code = 200;

